File: src/main/java/com/skylot222/service/UserRoleService.java
package com.skylot222.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylot222.domain.UserRole;
import com.skylot222.dto.UserRoleDTO;
import com.skylot222.dto.UserRoleSearchDTO;
import com.skylot222.dto.UserRolePageDTO;
import com.skylot222.dto.UserRoleConvertCriteriaDTO;
import com.skylot222.service.GenericService;
import com.skylot222.dto.common.RequestDTO;
import com.skylot222.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface UserRoleService extends GenericService<UserRole, Integer> {

	List<UserRole> findAll();

	ResultDTO addUserRole(UserRoleDTO userRoleDTO, RequestDTO requestDTO);

	ResultDTO updateUserRole(UserRoleDTO userRoleDTO, RequestDTO requestDTO);

    Page<UserRole> getAllUserRoles(Pageable pageable);

    Page<UserRole> getAllUserRoles(Specification<UserRole> spec, Pageable pageable);

	ResponseEntity<UserRolePageDTO> getUserRoles(UserRoleSearchDTO userRoleSearchDTO);
	
	List<UserRoleDTO> convertUserRolesToUserRoleDTOs(List<UserRole> userRoles, UserRoleConvertCriteriaDTO convertCriteria);

	UserRoleDTO getUserRoleDTOById(Integer userRoleId);







}

File: skylot222_ui/src/services/UserRoleService.js
import http from "../http-common"; 

class UserRoleService {
  getAllUserRoles(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/userRole/userRoles`, searchDTO);
  }

  get(userRoleId) {
    return this.getRequest(`/userRole/${userRoleId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/userRole?field=${matchData}`, null);
  }

  addUserRole(data) {
    return http.post("/userRole/addUserRole", data);
  }

  update(data) {
  	return http.post("/userRole/updateUserRole", data);
  }
  
  uploadImage(data,userRoleId) {
  	return http.postForm("/userRole/uploadImage/"+userRoleId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new UserRoleService();
